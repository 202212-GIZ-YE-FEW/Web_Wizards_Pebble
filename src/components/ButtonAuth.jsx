import Image from "next/image";
import { Button } from "react-flatifycss";

import "flatifycss/dist/css/flatify-min.css";
import styles from "./ButtonAuth.module.css";

const ButtonAuth = (props) => {
    const { icon, alt, width, height } = props
    return (
        <>
            <Button
                className={`w-full h-14 px-6 transition ${styles.ButtonAuth}`}
            >
                <div className='relative flex items-center space-x-4 justify-center'>
                    <Image
                        src={icon}
                        alt={alt}
                        width={width}
                        height={height}
                    />
                    <span className='block w-max font-medium tracking-wide text-gray-900 sm:text-base'>
                        {props.text}
                    </span>
                </div>
            </Button>
        </>
    );
};

export default ButtonAuth;
