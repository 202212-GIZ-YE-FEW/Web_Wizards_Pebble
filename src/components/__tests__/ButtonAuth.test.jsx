import renderer from "react-test-renderer";

import ButtonAuth from "../ButtonAuth";

it("renders correctly", () => {
    const tree = renderer.create(<ButtonAuth />).toJSON();
    expect(tree).toMatchSnapshot();
});
