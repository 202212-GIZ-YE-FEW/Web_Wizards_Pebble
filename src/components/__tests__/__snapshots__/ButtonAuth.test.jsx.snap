// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="button__ButtonWrapper-sc-1aw98b4-0 button w-full h-14 px-6 transition ButtonAuth"
  disabled={false}
>
  <div
    className="relative flex items-center space-x-4 justify-center"
  >
    <img
      data-nimg="1"
      decoding="async"
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src=""
      style={
        Object {
          "color": "transparent",
        }
      }
    />
    <span
      className="block w-max font-medium tracking-wide text-gray-900 sm:text-base"
    />
  </div>
</button>
`;
